src/components/screenSharing/services/screenSharingUiSrv.js

```diff
@@ -53,16 +54,43 @@
                     $animate.enter(screenSharingElement[0], screenSharingPhElement[0]);
                     $compile(screenSharingElement)(childScope);
                 });
+
+                return defer.promise;
             }
 
             this.activateScreenSharing = function (userSharingState) {
-                _activateScreenSharing(userSharingState);
+                return _activateScreenSharing(userSharingState);
             };
 
             this.endScreenSharing = function () {
                 _endScreenSharing();
             };
 
+            this.showScreenSharingConfirmationPopUp = function(){
+                var translationsPromMap = {};
+                translationsPromMap.title = $translate('SCREEN_SHARING.SHARE_SCREEN_REQUEST');
+                translationsPromMap.content= $translate('SCREEN_SHARING.WANT_TO_SHARE',{
+                    name: "Student/Teacher"
+                });
+                translationsPromMap.acceptBtnTitle = $translate('SCREEN_SHARING.REJECT');
+                translationsPromMap.cancelBtnTitle = $translate('SCREEN_SHARING.ACCEPT');
+                return $q.all(translationsPromMap).then(function(translations){
+                    var popUpInstance = PopUpSrv.warning(
+                        translations.title,
+                        translations.content,
+                        translations.acceptBtnTitle,
+                        translations.cancelBtnTitle
+                    );
+                    return popUpInstance.promise.then(function(res){
+                        return $q.reject(res);
```

Comment:
The pop up behaviour is opposite than what we need in screen sharing, so the answer is yes.